### ISSUE TITLE
files.getChunkedUploader is stuck on large files

### ISSUE THREAD

**Issue by elad-sf** [box/box-node-sdk#497]
- [v] I have checked that the [SDK documentation][sdk-docs] and [API documentation][api-docs] doesn't solve my issue

### Description of the Issue
When uploading large files (over 200MB from my testing) using the enterprise auth client and the chunked uploader, the upload is stuck in this process and the CPU goes up to 100% without ever coming down.
When uploading files with the chunked uploader that are smaller than 200MB (tested up to 150MB) all works as expected.
When changing the chunked uploader to the files.uploadFile API, all works as expected.
It seems like the chunked uploader is in infinite loop without ever throwing an exception or reaching timeout.

### Versions Used
Node SDK: 1.30.0 / 1.31.0 - tested both
Node.js: 12.14.0
OS:
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"

### Steps to Reproduce
```
const fs = require('fs');
const path = require('path');

const BoxSDK = require('box-node-sdk');

class BoxSDKClient {
    constructor(sdkOptions, enterpriseId, folderId) {
        const sdk = new BoxSDK(sdkOptions);
        this.boxClient = sdk.getAppAuthClient('enterprise', enterpriseId);
        this.folderId = folderId;
    }

    async uploadFile(folderName, fileName, filePath, fileSize) {
        try {
            let folderId = null;
            const folderItems = await this.boxClient.folders.getItems(this.folderId);
            for (let item of folderItems.entries) {
                if (item.name === folderName) {
                    folderId = item.id;
                }
            }
            if (!folderId) {
                const createdFolder = await this.boxClient.folders.create(this.folderId, folderName);
                folderId = createdFolder.id;
            }

            const fileStream = fs.createReadStream(filePath);
            console.log(`Starting the upload to box. file: ${fileName}, size: ${fileSize}`);
            const uploader = await this.boxClient.files.getChunkedUploader(folderId, fileSize, fileName, fileStream);
            const uploadedFile = await uploader.start();

            if (!uploadedFile) {
                throw new Error('Failed to upload file to box');
            }

            return this.boxClient.files.getDownloadURL(uploadedFile.entries[0].id);
        } catch (e) {
            console.error(`Failed to send file via Box`, e);
            return Promise.reject(e);
        }
    }
}

const main = async () => {
    const sdkOptions = {
        clientID: 'CLIENT_ID',
        clientSecret: 'CLIENT_SECRET',
        appAuth: {
            keyID: 'KEY_ID',
            privateKey: 'PRIVATE_KEY',
            passphrase: 'PASSPHRASE'
        }
    };
    const enterpriseId = 'ENTERPRISE_ID';
    const folderId = 'FOLDER_ID';

    console.log('logging in...');
    const boxClient = new BoxSDKClient(sdkOptions, enterpriseId, folderId);
    console.log('logged in successfully');

    const filePath = path.join(__dirname, 'LARGE_FILE_NAME');
    const fileStat = fs.statSync(filePath);
    
    try {
        await boxClient.uploadFile('FOLDER_NAME_ON_BOX', 'LARGE_FILE_NAME_ON_BOX', filePath, fileStat.size);
    } catch (err) {
        console.error('failed to upload file!', err);
    }
}

main();
```
Please try the code above when using enterprise auth client with files over 200MB.

### Error Message, Including Stack Trace
No error message is provided - the CPU goes up to 100% and stays there without the file being uploaded (tested on several different environments), only killing the process helps to free the CPU.

**Comment by sujaygarlanka** [comment]
@elad-sf I could not replicate the issue you were having. I used the following code to test. 
`
var sdkConfig = require('path-to-json-file');
var sdk = BoxSDK.getPreconfiguredInstance(sdkConfig);
var jwtClient = sdk.getAppAuthClient('enterprise');
var stat = fs.statSync('path-to-file');
var stream = fs.createReadStream('path-to-file');
console.log(stat.size)
jwtClient.files.getChunkedUploader('0', stat.size, 'large.pptx', stream)
	.then(uploader => uploader.start())
	.then(file => {
		console.log(file)
	});
`
It is from our docs and is just the chunked uploading part of your code. The issue could be with the how you instantiate your client. Take a look at the examples here on how to create your client (https://github.com/box/box-node-sdk/blob/master/docs/authentication.md#server-auth-with-jwt). Hope this helps!

**Comment by elad-sf** [comment]
@sujaygarlanka 
Thanks for the response, unfortunately this is not related to initialisation.
I was able to reproduce it in many environments but all of them had the same Node version..
After more investigation and profiling, this is a problem in the Node version of 12.14.0, which was already fixed in 12.14.1.
After upgrading, everything works as expected.
I'm writing it here so if anyone is having the same error will see the result.

So if you want you can add known issue, I'm closing this issue.
